Gems/ExpressionEvaluation/Code/Source/ElementInformationSerializer.inl

```diff
@@ -0,0 +1,181 @@
+
+        JsonSerializationResult::Result Store
+            ( rapidjson::Value& outputValue
+            , const void* inputValue
+            , const void* defaultValue
+            , const Uuid& valueTypeId, JsonSerializerContext& context) override
+        {
+            namespace JSR = JsonSerializationResult;
+
+            AZ_Assert(valueTypeId == azrtti_typeid<ElementInformation>(), "ElementInformation Store against value typeID that "
+                "was not ElementInformation");
+            AZ_Assert(inputValue, "ElementInformation Store against null inputValue pointer ");
+
+            auto inputScriptDataPtr = reinterpret_cast<const ElementInformation*>(inputValue);
+            auto defaultScriptDataPtr = reinterpret_cast<const ElementInformation*>(defaultValue);
+
+             if (defaultScriptDataPtr)
+             {
+                 if (inputScriptDataPtr->m_id == defaultScriptDataPtr->m_id
+                 && AZ::Helpers::CompareAnyValue(inputScriptDataPtr->m_extraStore, defaultScriptDataPtr->m_extraStore))
+                 {
+                     return context.Report
+                             ( JSR::Tasks::WriteValue, JSR::Outcomes::DefaultsUsed, "ElementInformation Store used defaults for "
+                                "ElementInformation");
+                 }
+             }
+
+            JSR::ResultCode result(JSR::Tasks::WriteValue);
+            outputValue.SetObject();
+
+            result.Combine(ContinueStoringToJsonObjectField
+                ( outputValue
+                , "Id"
+                , &inputScriptDataPtr->m_id
+                , defaultScriptDataPtr ? &defaultScriptDataPtr->m_id : nullptr
+                , azrtti_typeid<decltype(inputScriptDataPtr->m_id)>()
+                , context));
+
+            if (!inputScriptDataPtr->m_extraStore.empty())
+            {
+                rapidjson::Value typeValue;
+                result.Combine(StoreTypeId(typeValue, inputScriptDataPtr->m_extraStore.type(), context));
+                outputValue.AddMember
+                    ( rapidjson::StringRef(JsonSerialization::TypeIdFieldIdentifier)
+                    , AZStd::move(typeValue)
+                    , context.GetJsonAllocator());
+
+                result.Combine(ContinueStoringToJsonObjectField
+                    ( outputValue
+                    , "Value"
+                    , AZStd::any_cast<void>(const_cast<AZStd::any*>(&inputScriptDataPtr->m_extraStore))
+                    , defaultScriptDataPtr ? AZStd::any_cast<void>(const_cast<AZStd::any*>(&defaultScriptDataPtr->m_extraStore)) : nullptr
+                    , inputScriptDataPtr->m_extraStore.type()
+                    , context));
+
+            }
+            else
+            {
+                rapidjson::Value emptyAny;
+                AZStd::string emptyAnyName(EmptyAnyIdentifier);
+                emptyAny.SetString(emptyAnyName.c_str(), aznumeric_caster(emptyAnyName.size()), context.GetJsonAllocator());
```

Comment:
The temporary `AZStd::string` is a bit overkill since `EmptyAnyIdentifier` is a `const char*`. Instead `strlen` can be used to get the string length and avoid having to copy the identifier, or if it was upgraded to `AZStd::string_view` the length will be stored already.
Since the EmptyAnyIdentifier is static not using the temporary `AZStd::string` will also allow the identifier to be passed in by reference to avoid another string copy.

Reply:
X